### DISCUSSION TITLE
Unit conversion e.g. metric / imperial

### DISCUSSION THREAD

**Discussion by bishsbytes** [supabase/supabase#4938] [Questions]
Hi all,

I'm building an app that will allow users to store metrics like weight & distance. I want to ensure users can use the system of measurment they're most comfortable with i.e. imperial or metric.

I'm having trouble working out how to do this the 'Supabase way'. Normally, I'd just do on-the-fly conversions (e.g. kg -> lb) on the middleware and store the unit type & value in the DB.

I've had a go at using a Generated column with a conversion formula (i.e. take weight_kg, convert from kg to lb and create generated column weight_lb), but I think this will only support one way conversion from kg to lb and not the other way around.

Am I missing anything obvious?

Many thanks in advance!

**Comment by GaryAustin1** [comment]
One way is to use rpc functions to read and write the table.  Pass in values and a flag for metric or imperial write all data in metric can convert before writing if flag is imperial.  On rpc read function convert the single column if flag is imperial.

Another way, if you store both columns, you could use a trigger function on writes/updates, that depending on which column is passed in, writes the other column with the conversion factor.  So if they fill in weight_kg in the insert/update request you detect that is in the new variable and convert and fill in new.weight_lb.  The only issue with this is on updates they must null out1 of the columns, or you must compare the old values to see which one they changed and then update the other.  Update trigger has access to both old and new.

**Reply by bishsbytes** [reply]
Thanks @GaryAustin1 - I did think about triggers, but not about about RPC. Will experiment and see which works out.
